src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -1,18 +1,70 @@
 package wolox.training.controllers;
 
-import org.springframework.stereotype.Controller;
-import org.springframework.ui.Model;
+import java.util.List;
+import javax.validation.Valid;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.DeleteMapping;
 import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
 import org.springframework.web.bind.annotation.RestController;
+import wolox.training.exceptions.BookIdMismatchException;
+import wolox.training.exceptions.BookNotFoundException;
+import wolox.training.models.Book;
+import wolox.training.repositories.BookRepository;
 
-@Controller
+@RestController
+@RequestMapping(value = "/books", produces = MediaType.APPLICATION_JSON_VALUE)
 public class BookController {
 
-    @GetMapping("/greeting")
-    public String greeting(@RequestParam(name = "tunombre", required = false, defaultValue = "World") String name,
-        Model model) {
-        model.addAttribute("name", name);
-        return "greeting";
+    private final BookRepository bookRepository;
+
+    @Autowired
+    public BookController(BookRepository bookRepository) {
```

Comment:
This constructor is not required. With adding the @Autowired annotation on the attribute
`private BookRepository bookRepository` is sufficient.

Reply:
I think it is better to use constructor injection, because we can know what the dependencies of a class are. As for field injection, it is more difficult to know the dependencies of a class. In addition, field injection makes it difficult to perform unit tests because of all the complexity of reflection.

Reply:
ok great, as a comment I add that the tests can be done in the same way, if you inject the dependency in the attribute. Spring has tools like Mockito and others to perform your unit tests. This will be seen later in the training.